backend/pkg/table-dependency/table-dependency.go

```diff
@@ -336,7 +336,8 @@ func processCycles(
 			if isTableInCycles(cycles, fkTable) {
 				if len(fkCols.NullableColumns) > 0 {
 					updateConfig.appendDependsOn(fkTable, fkCols.NullableColumns)
-				} else {
+				}
+				if len(fkCols.NonNullableColumns) > 0 {
```

Comment:
This is the main bugfix here that I think is causing issues with circular dependencies.

This was resulting in a table that has a non-nullable dependency from being added as a dependency in the run order.
It is being added as a necessary foreignKey in the runconfig, but not as a dependency.
This results in both tables being synced at the same time, which causes FK constraint errors.